src/lib/loaders/batchLoader.ts

```diff
@@ -1,15 +1,57 @@
 import DataLoader from "dataloader"
-import { chain, flatten, chunk } from "lodash"
+import { chain } from "lodash"
 import config from "config"
 
 const { ENABLE_RESOLVER_BATCHING } = config
 
+interface NormalKey {
+  id: string
+  [key: string]: any
+}
+
+type Key = string | NormalKey
+
+type KeyGroupList = string[]
+
+interface CompactedKey {
+  id: string[]
+  [key: string]: any
+}
+
+interface GravityResult {
+  _id: string
+  _slug: string
+  [key: string]: any
+}
+
+/**
+ *  This function is used by the dataloader to determine the uniqueness of keys.
+ *  Because keys can be objects, it's important to be able to differentiate between
+ *  their contents. If an object is empty except for an id field, we just use id.
+ *  Otherwise we stringify the object so it can be compared via strict comparison.
+ */
+export const cacheKeyFn = (key: Key): string => {
+  if (typeof key === "object" && key !== null) {
```

Comment:
http://2ality.com/2013/04/quirk-undefined.html

Reply:
I'm not sure what this relates to. If the key is undefined, I'm just returning it like any other value. I'm not sure how to do error handling here, because if I throw an exception it's going to fail the entire query... maybe that's what we want though? Hmmm...